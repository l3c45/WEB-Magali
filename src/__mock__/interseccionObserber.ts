// import { vi } from "vitest";


// const intersectionObserverMock = () => ({
//   observe: () => null,
//   disconnect: () => null,

// });
// window.IntersectionObserver = vi.fn().mockImplementation(intersectionObserverMock)
  



import { vi } from 'vitest'

const IntersectionObserverMock = vi.fn(() => ({
  disconnect: vi.fn(),
  observe: vi.fn(),
  takeRecords: vi.fn(),
  unobserve: vi.fn(),
}))

vi.stubGlobal('IntersectionObserver', IntersectionObserverMock)


