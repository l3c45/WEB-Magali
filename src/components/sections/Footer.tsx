import { FC, useEffect, useRef, useState } from "react";
import useObserver from "../../hooks/useObserver";

const Footer: FC = () => {

  const footerRef = useRef(null);
  const observedElements = useObserver([footerRef]);

  return (
    <footer className="footer__container">
      <span
        data-ref="footer"
        ref={footerRef}
        className={observedElements.footer?.isInView ? "animate__animated animate__heartBeat" : ""}
      >

        Devloped by
        <a href="https://lucas-porfolio.vercel.app/" target={"_blank"}>
          Zarate Lucas
        </a>
      </span>
      <span> Copyright {new Date().getFullYear()} </span>
    </footer>
  );
};

export default Footer;
