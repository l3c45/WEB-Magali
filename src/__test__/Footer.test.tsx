import { render, screen } from '../utils/testUtils'
import '../__mock__/interseccionObserber'
import Footer from '../components/sections/Footer'


describe('Testing Footer Component', () => {
  beforeEach(() => {
    render(<Footer />)
  })

  it('Should render author name ', () => {
    const text = 'Zarate Lucas'

    expect(screen.getByText(text)).toMatchSnapshot()
  })

  it('Should render actual year ', () => {
    const year = String(new Date().getFullYear())
    const span = screen.getByText(/copyright/i)

    expect(span).toContainHTML(year)
  })

  it('Shouldnt have animate__animated class  in initial render ', () => {
    const span = screen.getByText(/developed/i)

    expect(span).not.toHaveClass('animate__animated')
  })
})
