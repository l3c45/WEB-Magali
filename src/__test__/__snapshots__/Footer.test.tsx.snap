// Vitest Snapshot v1

exports[`Testing Footer Component > Should render author name  1`] = `
<a
  href="https://lucas-porfolio.vercel.app/"
  target="_blank"
>
  Zarate Lucas
</a>
`;
